plugins/restful/RestfulEntityBase.php

```diff
@@ -698,7 +698,13 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missi
         $field_value = $this->propertyValuesPreprocess($property_name, $request[$public_field_name], $public_field_name);
       }
 
-      $wrapper->{$property_name}->set($field_value);
+      $sub_property = $info['sub_property'];
+      if (empty($sub_property)) {
+        $wrapper->{$property_name}->set($field_value);
+      }
+      else {
+        $wrapper->{$property_name}->{$sub_property}->set($field_value);
```

Comment:
@amitaibu is it correct or is it supposed to handled on a higher level?

Reply:
I believe looks right.